parsers/meta-yaml.h

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ *   Copyright (c) 2016, Masatake YAMATO
+ *   Copyright (c) 2016, Red Hat, K.K.
+ *
+ *   This source code is released for free distribution under the terms of the
+ *   GNU General Public License version 2 or (at your option) any later version.
+ *
+ */
+
+#ifndef CTAGS_YAML__H
+#define CTAGS_YAML__H
+
+#include "general.h"
+#include "types.h"
+
+#ifdef HAVE_LIBYAML
+#include <yaml.h>
+#else
+#define yaml_token_t void
+#endif
+
+typedef void (* yamlCallback) (yaml_token_t *token, void *data);
+
+struct yamlParserClient {
+	langType lang;
+	yamlCallback callback;
+	void* (* prepareForNewInput) (void);
+	void  (* teardownForTheLastInput) (void*);
```

Comment:
I'm not sure those callbacks are well named, at first I'd imagine they are called mostly like the parser's `initialize()` and `finalize()`, as in my head "last input" has to mean "last input file" or something, I didn't expect it to be the start and end of *each* input file.
Maybe `inputStart` and `inputEnd` or something?